thrust/testing/catch2_test_cuda_iterators.cu

```diff
@@ -245,3 +245,42 @@ TEST_CASE("transform_iterator", "[iterators]")
                  discard);
   }
 }
+
+struct AddZipped
+{
+  template <class T>
+  [[nodiscard]] _CCCL_HOST_DEVICE constexpr T operator()(cuda::std::pair<T, T> val) const noexcept
+  {
+    return val.first + val.second;
+  }
+};
+
+TEST_CASE("zip_iterator", "[iterators]")
+{
+  { // device system
+    thrust::device_vector<int> vec{-1, -1, -1, -1};
+    auto iter = cuda::make_transform_iterator(
+      cuda::make_zip_iterator(cuda::counting_iterator{0}, cuda::counting_iterator{4}), AddZipped{});
+    thrust::copy(iter, iter + 4, vec.begin());
+    thrust::device_vector<int> expected{4, 6, 8, 10};
+    CHECK(thrust::equal(vec.begin(), vec.end(), expected.begin()));
```

Comment:
Suggestion: could we include a host or device iterator in the zip iterator as well? This test only zips iterators with `any_system_tag`.